lib/generators/templates/setup/bin_setup.rb

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env ruby
+require "fileutils"
+require "open3"
+
+# path to your application root.
+APP_ROOT = File.expand_path("..", __dir__)
+
+def system!(*args)
+  system(*args, exception: true)
+end
+
+FileUtils.chdir APP_ROOT do
+  if ENV["CI"].nil?
+    puts "\n== Installing Homebrew Bundle from Brewfile =="
+    system! "brew bundle"
+
+    puts "\n== Starting postgresql@15 =="
+    system! "brew services restart postgresql@15"
+
+    puts "\n== Starting redis =="
+    system! "brew services restart redis"
+
+    _, _, status = Open3.capture3("which asdf")
+    if status.success?
+      puts "== Installing tool dependecies via asdf =="
+      system! "asdf plugin-add nodejs https://github.com/asdf-vm/asdf-nodejs"
+      system! "asdf plugin-add ruby https://github.com/asdf-vm/asdf-ruby"
+      system! "asdf plugin-add yarn https://github.com/twuni/asdf-yarn"
+      system! "asdf plugin-update --all"
+      system! "asdf install"
+    end
+  end
+
+  puts "== Installing dependencies =="
+  system! "gem install bundler --conservative"
+  system("bundle check") || system!("bundle install")
+  system("yarn install --check-files") if File.exist?("package.json")
```

Comment:
It's possible [this line will already exist](https://github.com/rails/rails/blob/f14401aa17c02cfad2213b5bb92a650f43ee4b86/railties/lib/rails/generators/rails/app/templates/bin/setup.tt#L21), so we might want to check for it first?

Reply:
Actually, that's not necessary since we're overriding the file and have a conditional.